scio-examples/src/main/scala/com/spotify/scio/examples/extra/ParquetExample.scala

```diff
@@ -108,21 +114,19 @@ object ParquetExample {
 
   private def avroSpecificIn(sc: ScioContext, args: Args): ClosedTap[String] = {
     // Macros for generating column projections and row predicates
-    val projection = Projection[Account](_.getId, _.getName, _.getAmount)
+    // account_status is the only field with default value that can be left out the projection
+    val projection = Projection[Account](_.getId, _.getType, _.getName, _.getAmount)
     val predicate = Predicate[Account](x => x.getAmount > 0)
 
     sc.parquetAvroFile[Account](args("input"), projection, predicate)
-      // The result Account records are not complete Avro objects. Only the projected columns are present while the rest are null.
-      // These objects may fail serialization and it’s recommended that you map them out to tuples or case classes right after reading.
-      .map(x => AccountProjection(x.getId, Some(x.getName.toString)))
```

Comment:
big behavioral breaking change here. We'll not allow construction of incomplete avro record.
IHMO since we have to map it anyway, user must use the typed api.